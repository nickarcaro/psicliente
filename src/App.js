import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
//importación de páginas
import Error404 from "./components/Error404/Error404";
import Dashboard from "./components/Dashboard/Dashboard";
import Login from "./components/Auth/Login/Login";
import SignUp from "./components/Auth/Signup/SignUp";
//importacion de contextos (hooks)
import AuthState from "./hooks/context/auth/authState";
import AlertaState from "./hooks/context/alertas/alertaState";
import tokenAuth from "./config/token";
import RutaPrivada from "./components/rutas/RutaPrivada";

// Revisar si tenemos un token
const token = localStorage.getItem("token");
if (token) {
  tokenAuth(token);
}

function App() {
  return (
    <AlertaState>
      <AuthState>
        <Router>
          <Switch>
            <Route exact path="/" component={Login} />
            <Route exact path="/signup" component={SignUp} />
            <RutaPrivada exact path="/home" component={Dashboard} />
            <Route component={Error404} />
          </Switch>
        </Router>
      </AuthState>
    </AlertaState>
  );
}

export default App;
