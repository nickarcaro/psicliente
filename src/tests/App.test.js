import "@testing-library/jest-dom";
import App from "../App";
import { shallow } from "enzyme";
import React from "react";

/* metodologia de las pruebas: 

-No se prueban los const, sino las funciones, clases (toda wa que retorne algo).

-  describe a√±ade la descripcion de la prueba
*/

describe("Prueba de <App/>", () => {
  test("debe mostrarse <App/> Correctamente", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
