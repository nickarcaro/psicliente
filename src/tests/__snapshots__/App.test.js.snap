// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <App/> debe mostrarse <App/> Correctamente 1`] = `
<BrowserRouter>
  <Switch>
    <RouteWithSubRoutes
      component={[Function]}
      exact={false}
      key="0"
      path="/"
      routes={
        Array [
          Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          Object {
            "component": [Function],
            "exact": true,
            "path": "/contacto",
          },
          Object {
            "component": [Function],
          },
        ]
      }
    />
  </Switch>
</BrowserRouter>
`;
